components/Form/TextInput/index.js

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+
+const TextInput = ({ label, name, onChange, validate }) => {
+  const [inputValue, setInputValue] = useState('');
```

Comment:
I think this is cool for now, but I wonder if having the value as state locally within the `TextInput` component will cause us problems if we need to pass a value into this in the future? You usually see `value` kept as state in the parent and then just passed in as a prop. I see how that would make your validation more difficult though - so maybe we should just cross that bridge when/if we get to it.